src/components/Buy/BuyInput.js

```diff
@@ -104,17 +100,40 @@ const BuyInput = observer((props) => {
     };
 
     const checkActive = () => {
-        console.log('active', {
-            buyAmount: tradingStore.buyAmount,
-            account,
-            active: tradingStore.buyAmount > 0 && !!account && !hasError,
-        });
         return tradingStore.buyAmount > 0 && !!account && !hasError;
     };
 
-    const validateNumber = (value) => {
+    const validateNumber = async (value) => {
         tradingStore.setBuyAmount(value);
         hasError = !(value > 0);
+        const status = validateTokenValue(value);
+
+        /*
+            So, you need to make sure the PRICE is loaded and UPDATED
+            So we don't do it here, when get get a new PRICE - we check if the input is valid, and if it is, we refresh.
+
+            We also need to update WHEN the input is changed & valid - if there is a PRICE loaded, set it.
+
+            When a PRICE is loaded, set the result based on input
+            When a INPUT is added, set the results based on input
```

Comment:
Looks like could probably clean up this comment

Reply:
Tru